### PR TITLE
gh-94795: Remove `#ifdef __cplusplus` from C-only code

### PR DESCRIPTION
**Pull request by arhadthedev**
The PR contains one-type replacements so I don't know how to split them into smaller PRs.

Edit: also removed a dead branch from: https://github.com/python/cpython/blob/cceac5dd06fdbaba3f45b8be159dfa79b74ff237/Python/getcompiler.c#L15-L19

* Issue: gh-94795

### CODE REVIEW

**Comment by encukou** [comment]
Does this solve an issue?
We don't generally do style-only changes.

**Comment by ericsnowcurrently** [comment]
IIRC, @gpshead & @Yhg1s have had some interest in this ifdef in the past.

File: Modules/_ctypes/ctypes_dlfcn.h [hunk 1]
```diff
@@ -1,10 +1,6 @@
 #ifndef _CTYPES_DLFCN_H_
 #define _CTYPES_DLFCN_H_
 
-#ifdef __cplusplus
-extern "C" {
-#endif /* __cplusplus */
-
```
Comment (arhadthedev) [inline Modules/_ctypes/ctypes_dlfcn.h:7]: Area of doubt 1.

**Comment by arhadthedev** [comment]
I restored all `*.h` files back because [CPython has a few C++ modules](https://github.com/python/cpython/search?l=C%2B%2B&q=%2A.cpp+in%3Apath) so even private headers must be aware of them.

To reiterate, currently this PR removes chunks of totally dead code because `__cplusplus` is never defined for `*.c` files

**Comment by arhadthedev** [comment]
Closing because nobody evaluated the change as non-cosmetical.

**Pull request closed without merging**